pkg/scheduler/framework/plugins/helper/node_affinity.go

```diff
@@ -72,7 +72,9 @@ func PodMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 // nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
-	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
-		"metadata.name": node.Name,
-	})
+	matches, err := corev1.MatchNodeSelectorTerms(node, nodeSelectorTerms)
+	if err != nil {
+		klog.ErrorS(err, "Couldn't parse all NodeSelector terms", "terms", nodeSelectorTerms)
```

Comment:
if a selector has a parse error, and we evaluate it many times against lots of nodes, won't this spam the logs terribly?

Reply:
What about to inline `nodeMatchesNodeSelectorTerms` into `PodMatchesNodeSelectorAndAffinityTerms` so you can also signal which pod has the issue?

Reply:
> if a selector has a parse error, and we evaluate it many times against lots of nodes, won't this spam the logs terribly?

Does anyone recall what is the reason why node affinity terms are not properly validated on the apiserver side?

Reply:
> Does anyone recall what is the reason why node affinity terms are not properly validated on the apiserver side?

This was more noticeable with pod affinity (https://github.com/kubernetes/kubernetes/pull/92714) but essentially we don't do any validation on these fields, and it can't be added because the tightened validation could break compatibility with existing objects